src/main/java/org/folio/circulation/services/LostItemFeeChargingService.java

```diff
@@ -112,6 +122,8 @@ private CompletableFuture<Result<Loan>> applyFees (ReferenceDataContext referenc
     return fetchFeeFineOwner(referenceData)
     .thenApply(this::refuseWhenFeeFineOwnerIsNotFound)
     .thenComposeAsync(this::fetchFeeFineTypes)
+    .thenApply(this::buildActualCostRecord)
+    .thenCompose(r -> r.after(this::createActualCostRecord))
```

Comment:
Is actual cost records built and saved here unconditionally, in any case? I don't see any checks here, neither I see them in `buildActualCostRecord` and `buildActualCostRecord` methods.

Reply:
The logic is changed, thanks.